contracts/IStaking.sol

```diff
@@ -1,24 +1,15 @@
-pragma solidity 0.5.0;
+pragma solidity ^0.8.24;
 
 /**
  * @title Staking interface, as defined by EIP-900.
  * @dev https://github.com/ethereum/EIPs/blob/master/EIPS/eip-900.md
  */
-contract IStaking {
-    event Staked(address indexed user, uint256 amount, uint256 total, bytes data);
-    event Unstaked(address indexed user, uint256 amount, uint256 total, bytes data);
-
+interface IStaking {
     function stake(uint256 amount, bytes calldata data) external;
-    function stakeFor(address user, uint256 amount, bytes calldata data) external;
```

Comment:
Any reason you removed the parameter name?

Reply:
hmm didn't mean to remove here. Adding it back